### PR TITLE
feat: manage recording database

### LINKED ISSUE #446: Manage Recording DB
### Feature request

I'd like a way to list all recordings and delete specific ones, from CLI. Currently, the #300 PR has methods for displaying all replays, but a way to do this through CLI would be beneficial.

usage could be something like
`python -m openadapt.db.list`
```
> [2] recording_2 : <timestamp> ...
> [1] test: <timestamp> ...
```
`python -m openadapt.db.remove 2`
```
> are you sure you want to delete recording_2? (y/n)
```
`y`
```
> deleted recording_2
```
`python -m openadapt.db.list`
```
> [1] test: <timestamp> ...
```

**There should also be a way to remove the latest recording by default**

In the future, maybe we can have a `backup` and `add` command too, to add recordings similar to #223 
### Motivation

I have a *broken* recording, which makes my latest visualization fail, but the previous one works. I'd like to remove this recording while still preserving my working one. This may be useful for training later.

### PR DESCRIPTION
**Pull request by 0dm**
**What kind of change does this PR introduce?**

implements new features:
`openadapt.db.list`
`openadapt.db.remove`

**Summary**

I moved crud.py & db.py to a new directory, db. Does this make sense @abrichr ? I can move them back if needed. 
In the package is remove.py, which can remove all recordings, or the latest, or a specified one by index. list.py will just 
list and number each recording, from latest to oldest. 

**Motivation**
- resolves #446 

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have performed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**
ex.
run `python -m openadapt.db.list`:

image

run `python -m openadapt.db.remove 4` and enter `y`

image

### CODE REVIEW

File: alembic/env.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 from alembic import context
 from openadapt.config import DB_URL
-from openadapt.db import Base
+from openadapt.db.db import Base
```
Comment (Mustaballer) [inline alembic/env.py:12]: I'm not sure if I like `.db.db`. Maybe we can do `from openadapt.db import db`.

File: openadapt/db/remove.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+def main() -> int:
+    """Removes a recording from the database."""
+    recordings = get_all_recordings()
+
+    if not recordings:
+        print("No recordings found.")
+        return 1
+
+    if len(sys.argv) != 2:
+        print("Usage: openadapt.db.remove <recording_index|latest|all>")
```
Comment (Mustaballer) [inline openadapt/db/remove.py:17]: is it recording_index or recording_id?

**Review by Mustaballer — APPROVED** [review]
Just had some minor thoughts but lgtm! This is definitely useful for https://github.com/OpenAdaptAI/OpenAdapt/pull/223

File: openadapt/db/remove.py (hunk 2 above)
Comment (0dm) [reply openadapt/db/remove.py:17]: ah should be index

File: alembic/env.py (hunk 1 above)
Comment (0dm) [reply alembic/env.py:12]: oh true, vscode did these changes for me

**Comment by 0dm** [comment]
@Mustaballer @abrichr  I think having a export / add feature to this would be nice, how can we integrate this with magic-wormhole ? I was thinking that we could do something like
```py
python -m openadapt.db.add <wormhole code|file_name> 
```
```py 
python -m openadapt.db.export <recording_id>
```
with options for local file or magic wormhole
```py 
python -m openadapt.db.add 
[1] magic wormhole
[2] browse local
```
I can implement a file picker with nicegui

**Comment by Mustaballer** [comment]
We can certainly do that, I do use two seperate functions for sending and receiving recordings in `share.py`(`send_recording`, `receive_recording`) in https://github.com/OpenAdaptAI/OpenAdapt/pull/223.

File: openadapt/db/remove.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+from openadapt.db.crud import delete_recording, get_all_recordings
+
+
+def main() -> int:
+    """Removes a recording from the database."""
+    recordings = get_all_recordings()
+
+    if not recordings:
+        print("No recordings found.")
```
Comment (abrichr) [inline openadapt/db/remove.py:13]: Please use `loguru`

File: openadapt/db/remove.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        return 1
+
+    if len(sys.argv) != 2:
```
Comment (abrichr) [inline openadapt/db/remove.py:16]: Please use `fire`

Comment (0dm) [reply openadapt/db/remove.py]: used click, see [ee15fd4](https://github.com/OpenAdaptAI/OpenAdapt/pull/451/commits/ee15fd4d0344eb0ed2ebf74306324bede1eeebb9)

**Pull request merged**